src/helix-ui/elements/HXFileTileElement.js

```diff
@@ -13,48 +16,94 @@ export class HXFileTileElement extends HXElement {
 
     $onCreate () {
         this._onDismiss = this._onDismiss.bind(this);
+        this._onRetry = this._onRetry.bind(this);
     }
 
     $onConnect () {
+        this.$upgradeProperty('details');
+        this.$upgradeProperty('extension');
+        this.$upgradeProperty('href');
         this.$upgradeProperty('icon');
         this.$upgradeProperty('name');
-        this.$upgradeProperty('href');
+        this.$upgradeProperty('progress');
+        this.$upgradeProperty('truncated');
+        this.$upgradeProperty('valid');
         this._btnDismiss.addEventListener('click', this._onDismiss);
+        this._btnRetry.addEventListener('click', this._onRetry);
     }
 
     $onDisconnect () {
         this._btnDismiss.removeEventListener('click', this._onDismiss);
+        this._btnRetry.removeEventListener('click', this._onRetry);
     }
 
     static get $observedAttributes () {
         return [
+            'details',
             'href',
             'icon',
             'name',
+            'progress',
         ];
     }
 
     $onAttributeChange (attr, oldVal, newVal) {
         switch (attr) {
+            case 'details':
+                this._elDetails.innerText = newVal;
+                break;
+
+            case 'href':
+                this._attrHrefUpdate(oldVal, newVal);
+                break;
+
             case 'icon':
                 this._elIcon.type = newVal;
                 break;
 
-            case 'href':
-                if (newVal !== null) {
-                    this._elLink.href = newVal;
+            case 'name':
+                this._elExt.innerText = this.extension;
+
+                if (this.truncated) {
+                    this._renderName();
                 } else {
-                    this._elLink.removeAttribute('href');
+                    this._elName.innerText = this.name;
                 }
+
                 break;
 
-            case 'name':
-                this._elLink.innerText = newVal;
-                this._elExt.innerText = this._extension || '';
+            case 'progress':
+                this._elProgress.value = newVal;
                 break;
         }
     }
 
+    /**
+     * https://regex101.com/r/K8XCbn/2
+     * @readonly
+     * @type {String}
+     */
+    get extension () {
+        let re = /(?:\.([^.]+))?$/;
+        return re.exec(this.name)[1] || '';
+    }
+
+    /**
+     * URL to download the file.
+     *
+     * @type {String}
+     */
+    get href () {
+        return this.getAttribute('href');
+    }
+    set href (newVal) {
+        if (newVal === null) {
```

Comment:
supports `el.href = null` to remove built-in link behavior via javascript property